### PR TITLE
Use node suggestion API in node autocomplete UI

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Call node suggestion API from node autocomplete UI.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs [hunk 1]
```diff
@@ -48,7 +49,9 @@ public class DynamoGeometryModel3D : MaterialGeometryModel3D
     {
         protected override SceneNode OnCreateSceneNode()
         {
-            return new DynamoMeshNode();
+            var node = new DynamoMeshNode();
+            node.CullMode = CullMode.Back;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs:53]: hmm any comments here?

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs]: Reverted. This change was unrelated.

File: src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs [hunk 2]
```diff
@@ -32,5 +32,16 @@ internal void InitializeDefaultAutoCompleteCandidates()
             }
             FilteredResults = candidates;
         }
+
+        internal void PopulateAutoCompleteCandidates()
+        {
+            var searchElements = PortViewModel.GetMatchingNodes();
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs:38]: Thanks for moving it. But I was hoping we can move all of the code from `PortViewModel` to the customized search view model here given that 1. those code are not really related to PortView 2. I feel `PortViewModel` does not necessarily obtain knowledge about search elements 3. We still have access to DynamoViewModel from here. What do you think?

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs:38]: I haven't moved the code yet. Doing so now.

File: src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs [hunk 3]
```diff
@@ -121,7 +121,8 @@ private void OnNodeAutoCompleteSearchControlVisibilityChanged(object sender, Dep
             Dispatcher.BeginInvoke(new Action(() =>
             {
                 SearchTextBox.Focus();
-                ViewModel.InitializeDefaultAutoCompleteCandidates();
+                //ViewModel.InitializeDefaultAutoCompleteCandidates();
+                ViewModel.PopulateAutoCompleteCandidates();
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs:125]: We probably want to consider returning the number of filtered results here, if the number is 0 which means we failed to find any match, then we still call `InitializeDefaultAutoCompleteCandidates()`.

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs:125]: The reason I commented this out was I noticed issues with this. On a mouse up, the node suggestion list was reverting to this default list each time.
